lib/blockscore/base.rb

```diff
@@ -66,18 +67,20 @@ def capture_attributes(source)
       @attributes = source.attributes
     end
 
-    def add_accessor(symbol, *_args)
+    def add_accessor(symbol)
       singleton_class.instance_eval do
         define_method(symbol) do
-          wrap_attribute(attributes[symbol])
+          wrap_attribute(attributes.fetch(symbol))
         end
       end
     end
 
-    def add_setter(symbol, *_args)
+    def add_setter(symbol)
+      attr = symbol.to_s.chop.to_sym
+      assert_mutable(attr)
       singleton_class.instance_eval do
```

Comment:
Not in this PR, but I definitely think we should examine this block. By modifying the singleton like this it is going to blow the global method cache. This affects everything in the whole runtime, not just this specific code.

A better approach would be to define the attributes up-front per-class, and then create the methods when the class is initialized.